### PR TITLE
Fix deadlock in updateRepository

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fix a potential deadlock in `updateRepository()`. The call to `UpdateSize()` previously did not use the provided session (i.e. the `e` parameter), and the resulting query could become blocked by the ongoing session/transaction, since both the query and the ongoing transaction need to access the `repository` table.

I experienced this deadlock locally, using PostgreSQL.

### CODE REVIEW

**Comment by sapk** [comment]
~~repo.UpdateSize is used in others places : like https://github.com/go-gitea/gitea/blob/be6edaddcba1e607884539a250bd19debabba751/models/repo_mirror.go#L151 or https://github.com/go-gitea/gitea/blob/930d1759aea9777767fcaf025f25076420c137c6/models/update.go#L103~~

Edit: Sorry, I missed some code part.

**Comment by sapk** [comment]
LGTM

File: models/repo.go [hunk 1]
```diff
@@ -619,18 +619,22 @@ func (repo *Repository) IsOwnedBy(userID int64) bool {
...
-func (repo *Repository) UpdateSize() error {
+func (repo *Repository) updateSize(e Engine) error {
 	repoInfoSize, err := git.GetRepoSize(repo.RepoPath())
 	if err != nil {
 		return fmt.Errorf("UpdateSize: %v", err)
 	}
 
 	repo.Size = repoInfoSize.Size + repoInfoSize.SizePack
-	_, err = x.ID(repo.ID).Cols("size").Update(repo)
+	_, err = e.Id(repo.ID).Cols("size").Update(repo)
```
Comment (lunny) [inline models/repo.go:629]: `ID` is better than `Id`

**Comment by lunny** [comment]
A small change required, otherwise LGTM

File: models/repo.go (hunk 1 above)
Comment (ethantkoenig) [reply models/repo.go:629]: `ID` is not in the `Engine` interface.

Comment (lunny) [reply models/repo.go:629]: OK. Let's do that later!

**Pull request merged**